### ISSUE TITLE
[Release] v0.8 Release Planning

### ISSUE THREAD

**Issue by junrushao1994** [apache/tvm#8976]
Thanks to everyone who contributed in the past cycle! We would love to use this thread to plan for the v0.8 release. The community has always been focusing on high quality releases. This process can take several weeks according to past releases, to allow us to collectively work together on items and track things that needs to be merged.

In the next few days we will be populating the release note in this thread. Most release note content will be derived from our monthly report.

@vinx13 and I volunteered to manage the release, and @icemelon and @yzhliu will mentor us as PMC members.

Please don't hesitate to share your thoughts :-)

**Comment by leandron** [comment]
Thanks for doing this work! There were some items listed on #7434, perhaps we can start with those?

**Comment by Hzfengsy** [comment]
Thanks for the work. I believe v0.8 is a good chance to land TensorIR scheduling (https://github.com/apache/tvm/issues/7527). Also, I will try my best to contribute some initial TensorIR tutorials and documentations before the v0.8 release.

**Comment by comaniac** [comment]
Agree with @leandron that we could firstly refer to the items there. Many "initial" features in v0.7 are now stable. For example:

* Initial automatic scheduling support -> stable.
* Initial command line driver interface -> stable.
* Intial Hexagon support -> stable.
* Bring your own codegen (BYOC) support -> now we have several backends.
  * [stable] NVIDIA TensorRT, Xilinx Vitis-AI, ARM compute library, ARM Ethos-N, etc.
  * [experimental] TBA.

**Comment by junrushao1994** [comment]
I wanted to propose as a highlight the TE-level auto-differentiation work, lead by @yzhliu, which unlocks TE-level training capability in the TVM stack

**Comment by tristanqiu8** [comment]
Expecting the new release!

**Comment by FrozenGene** [comment]
> Agree with @leandron that we could firstly refer to the items there. Many "initial" features in v0.7 are now stable. For example:
> 
>     * Initial automatic scheduling support -> stable.
> 
>     * Initial command line driver interface -> stable.
> 
>     * Intial Hexagon support -> stable.
> 
>     * Bring your own codegen (BYOC) support -> now we have several backends.
>       
>       * [stable] NVIDIA TensorRT, Xilinx Vitis-AI, ARM compute library, ARM Ethos-N, etc.
>       * [experimental] TBA.

Does our hexagon support is stable now? I am not sure about it. As I saw we still pull requests actively(like https://github.com/apache/tvm/pull/8986 to support model lauch). I think the status maybe is still not stable. @kparzysz-quic should give more definitive answer

**Comment by kparzysz-quic** [comment]
Hexagon support still needs more work.

**Comment by areusch** [comment]
Just checking--by "stable" for the "command line driver interface," does that mean "someone should write a script and presume we won't change the command syntax?"

For µTVM we could add:
- microTVM Project API -> initial support
- microTVM AOT executor -> initial support
- microTVM platforms: Zephyr, Arduino -> initial support

I'd like us to avoid using "uTVM" to refer to microTVM in the release notes and just spell out the word "micro" instead. While u is still a common abbreviation for mu, it's not necessarily widely known. We've renamed most occurrences of uTVM inside the codebase to avoid confusion.

**Comment by junrushao1994** [comment]
@vinx13 and I spent the previous 2 weeks categorizing the 69 pages of changelogs via git commits: https://docs.google.com/document/d/1D2rig32yZ4H48_tXHb_svnPnt4-FRx3JUbBxKyrW_-I/

Below are the categories we are using:

- TE & TIR & TIR Scheduling & TVM Script
- Relay IR
- AutoTVM & Auto Scheduler & Meta Schedule
- Training
- TOPI & Operators
- Executor & AOT & Profiler
- MicroTVM
- Arithmetic Analysis
- Frontend - TensorFlow & TFLite
- Frontend - PyTorch
- Frontend - ONNX
- Frontend - MXNet
- Frontend - Misc
- Codegen/Runtime: LLVM & ROCm
- Codegen/Runtime: Vulkan & SPIR-V
- Codegen/Runtime: OpenCL & VTA
- Codegen/Runtime: CUDA
- Codegen/Runtime: Arm
- Codegen/Runtime: Hexagon
- Codegen/Runtime: WASM
- Codegen/Runtime: C
- Codegen/Runtime: Metal
- BYOC
- CI & Docker & Build & Test
- TVMC
- Rust and Other Language Bindings
- Docs & Tutorials
- RPC
- Target
- Misc
- Community

Please feel free to suggest in the doc if there is any mistake :-)

**Comment by jiangjiajun** [comment]
Hi, @junrushao1994
I think we could add paddlepaddle frontend as a new feature in the release note of v0.8

**Comment by junrushao1994** [comment]
@jiangjiajun Sure! We will list experimental paddlepaddle frontend support as a separate category and a highlight of this release

**Comment by junrushao1994** [comment]
@Mousius Thanks for asking!

> does this mean that 0.8 will go out with half finished implementations for things, such as library integrations (i.e. CMSIS-NN) and tvmc arguments (tvmc is not yet stable as there's breaking changes incoming)

Yes, we directly cut main into the v0.8 branch: https://github.com/apache/tvm/tree/v0.8. We do not have to announce the features that are not complete yet, and if there is any major bugfixes, it should be backported to the v0.8 branch.

> can we rapidly move to focusing on a 0.9 release with a list of issues as suggested by @areusch so we can work towards stability in a 1.x release.

I agree. Hopefully we will have a more automatic release process with the help from @denise-k :-)

**Comment by junrushao1994** [comment]
Hi all, we cut a v0.8 release branch for Apache TVM: https://github.com/apache/tvm/tree/v0.8. Please find:
- The release note (candidate): https://github.com/apache/tvm/issues/9416
- The full changelog (candidate): https://gist.github.com/junrushao1994/c669905dbc41edc2e691316df49d8562

There have been tens of commits since last time we categorized the commits, and @vinx13 is actively working on adding them into our changelog.

Please let us know if we missed anything or need any tweak on wording/categorization on the release note!

**Comment by junrushao1994** [comment]
Also, if there is any bug/issue blocking the release, please don't hesitate to let us know in this thread :-)

**Comment by junrushao1994** [comment]
Blocker: We need this bugfix in to address a regression https://github.com/apache/tvm/pull/9421

**Comment by Lyken17** [comment]
Blocker: Conv2dTranspose currently does not support `groups`. But `groups` is popularly used in recent efficient models and Conv2dTranspose is an important operator in GAN related applications. 

* `groups` support: https://github.com/apache/tvm/issues/8182 https://github.com/apache/tvm/pull/8799
* `shape` check: https://github.com/apache/tvm/pull/9465

Will try my best to fix the problem this week.

**Comment by junrushao1994** [comment]
The release is cut and is available for test in https://github.com/apache/tvm/tree/v0.8

**Comment by junrushao1994** [comment]
Blocker #9486

**Comment by gyzou** [comment]
When would 0.8 official release be? Estimatedly, days, weeks, months?

**Comment by junrushao1994** [comment]
The release candidate v0.8.rc0 is approved:
* Voting thread: https://github.com/apache/tvm/issues/9504
* Voting result: https://github.com/apache/tvm/issues/9566
